runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -572,10 +597,32 @@ static StreamingDataflowWorker forTesting(
   private static void onPipelineConfig(
       StreamingEnginePipelineConfig config,
       Consumer<ImmutableSet<HostAndPort>> consumeWindmillServiceEndpoints,
-      AtomicInteger maxWorkItemCommitBytes) {
-    if (config.maxWorkItemCommitBytes() != maxWorkItemCommitBytes.get()) {
-      LOG.info("Setting maxWorkItemCommitBytes to {}", maxWorkItemCommitBytes);
-      maxWorkItemCommitBytes.set((int) config.maxWorkItemCommitBytes());
+      AtomicReference<OperationalLimits> operationalLimits) {
+    if (config.maxWorkItemCommitBytes() != operationalLimits.get().maxWorkItemCommitBytes) {
```

Comment:
Create an operationalLimits from all of the set values and then install it with a single set

A missing config value should probably revert back to the default not leave it as the last set value.  But if you are creating a new operationallimits each time I think that would be the case anwyay

Reply:
Let me know if this is what you intended. I'm not just setting the `OperationalLimits` based on whatever was set in the `StreamingEnginePipelineConfig`. Any values not explicitly set by the caller will use what ever the config's default values are. So at least in this block of code, the `OperationalLimit`'s default values never come into play.

The `StreamingEnginePipelineConfig` isn't a proto, so I can't tell if a value was explicitly set or not (there's no `hasFoo()` methods).